python/GafferSceneUI/SetEditor.py

```diff
@@ -0,0 +1,379 @@
+				"active" : len( selectedSetNames ) > 0,
+				"shortCut" : "Ctrl+Shift+C"
+			}
+		)
+
+		menuDefinition.append(
+			"/Select Set Members",
+			{
+				"command" : Gaffer.WeakMethod( self.__selectSetMembers ),
+				"active" : len( selectedSetNames ) > 0,
+			}
+		)
+
+		self.__contextMenu = GafferUI.Menu( menuDefinition )
+		self.__contextMenu.popup( widget )
+
+		return True
+
+	def __copySelectedSetNames( self, *unused ) :
+
+		if self.__plug is None :
+			return
+
+		selection = self.__selectedSetNames()
+		if len( selection ) > 0 :
+			data = IECore.StringVectorData( selection )
+			self.__plug.ancestor( Gaffer.ApplicationRoot ).setClipboardContents( data )
+
+	def __getSetMembers( self, *unused ) :
+
+		if self.__plug is None :
+			return
+
+		result = IECore.PathMatcher()
+		with Gaffer.Context( self.getContext() ) :
+			for setName in self.__selectedSetNames() :
+				result.addPaths( self.__plug.set( setName ).value )
+
+		return result
+
+	def __selectSetMembers( self, *unused ) :
+
+		locations = self.__getSetMembers()
+		if not locations.isEmpty() :
+			ContextAlgo.setSelectedPaths( self.getContext(), locations )
+
+	def __copySetMembers( self, *unused ) :
+
+		if self.__plug is None :
+			return
+
+		locations = self.__getSetMembers()
+		if not locations.isEmpty() :
+			data = IECore.StringVectorData( locations.paths() )
+			self.__plug.ancestor( Gaffer.ApplicationRoot ).setClipboardContents( data )
+
+GafferUI.Editor.registerType( "SetEditor", SetEditor )
+
+##########################################################################
+# _SearchFilterWidget
+##########################################################################
+
+class _SearchFilterWidget( GafferUI.PathFilterWidget ) :
+
+	def __init__( self, pathFilter ) :
+
+		self.__patternWidget = GafferUI.TextWidget()
+		GafferUI.PathFilterWidget.__init__( self, self.__patternWidget, pathFilter )
+
+		self.__patternWidget._qtWidget().setPlaceholderText( "Filter..." )
+
+		self.__patternWidget.editingFinishedSignal().connect( Gaffer.WeakMethod( self.__patternEditingFinished ), scoped = False )
+		self.__patternWidget.dragEnterSignal().connectFront( Gaffer.WeakMethod( self.__dragEnter ), scoped = False )
+		self.__patternWidget.dragLeaveSignal().connectFront( Gaffer.WeakMethod( self.__dragLeave ), scoped = False )
+		self.__patternWidget.dropSignal().connectFront( Gaffer.WeakMethod( self.__drop ), scoped = False )
+
+		self.__context = None
+		self.__scene = None
+
+		self._updateFromPathFilter()
+
+	def setContext( self, context ) :
+
+		self.__context = context
+
+	def setScene( self, scene ) :
+
+		self.__scene = scene
+
+	def _updateFromPathFilter( self ) :
+
+		self.__patternWidget.setText( self.pathFilter().getMatchPattern() )
+
+	def __patternEditingFinished( self, widget ) :
+
+		self.pathFilter().setMatchPattern( self.__patternWidget.getText() )
+
+	def __dragEnter( self, widget, event ) :
+
+		self.__patternWidget.setHighlighted( True )
```

Comment:
We need to check that `event.data` is non-empty StringVectorData before accepting here. Otherwise it looks like we accept _anything_, which is misleading.

Reply:
Fixed in 7c496f0.